programs/vote/src/vote_state/handler.rs

```diff
@@ -0,0 +1,314 @@
+            TargetVoteState::V3(v3) => v3.node_pubkey = node_pubkey,
+        }
+    }
+
+    pub fn votes(&self) -> &VecDeque<LandedVote> {
+        match &self.target_state {
+            TargetVoteState::V3(v3) => &v3.votes,
+        }
+    }
+
+    pub fn votes_mut(&mut self) -> &mut VecDeque<LandedVote> {
+        match &mut self.target_state {
+            TargetVoteState::V3(v3) => &mut v3.votes,
+        }
+    }
+
+    pub fn set_votes(&mut self, votes: VecDeque<LandedVote>) {
+        match &mut self.target_state {
+            TargetVoteState::V3(v3) => v3.votes = votes,
+        }
+    }
+
+    pub fn contains_slot(&self, slot: Slot) -> bool {
+        match &self.target_state {
+            TargetVoteState::V3(v3) => v3.contains_slot(slot),
+        }
+    }
+
+    pub fn last_voted_slot(&self) -> Option<Slot> {
+        match &self.target_state {
+            TargetVoteState::V3(v3) => v3.last_voted_slot(),
+        }
+    }
+
+    pub fn compute_vote_latency(voted_for_slot: Slot, current_slot: Slot) -> u8 {
+        std::cmp::min(current_slot.saturating_sub(voted_for_slot), u8::MAX as u64) as u8
+    }
+
+    pub fn root_slot(&self) -> Option<Slot> {
+        match &self.target_state {
+            TargetVoteState::V3(v3) => v3.root_slot,
+        }
+    }
+
+    pub fn set_root_slot(&mut self, root_slot: Option<Slot>) {
+        match &mut self.target_state {
+            TargetVoteState::V3(v3) => v3.root_slot = root_slot,
+        }
+    }
+
+    pub fn current_epoch(&self) -> Epoch {
+        match &self.target_state {
+            TargetVoteState::V3(v3) => v3.current_epoch(),
+        }
+    }
+
+    pub fn epoch_credits_last(&self) -> Option<&(Epoch, u64, u64)> {
+        match &self.target_state {
+            TargetVoteState::V3(v3) => v3.epoch_credits.last(),
+        }
+    }
+
+    pub fn credits_for_vote_at_index(&self, index: usize) -> u64 {
+        match &self.target_state {
+            TargetVoteState::V3(v3) => v3.credits_for_vote_at_index(index),
+        }
+    }
+
+    pub fn increment_credits(&mut self, epoch: Epoch, credits: u64) {
+        match &mut self.target_state {
+            TargetVoteState::V3(v3) => v3.increment_credits(epoch, credits),
+        }
+    }
+
+    pub fn process_timestamp(&mut self, slot: Slot, timestamp: i64) -> Result<(), VoteError> {
+        match &mut self.target_state {
+            TargetVoteState::V3(v3) => v3.process_timestamp(slot, timestamp),
+        }
+    }
+
+    pub fn process_next_vote_slot(
+        &mut self,
+        next_vote_slot: Slot,
+        epoch: Epoch,
+        current_slot: Slot,
+    ) {
+        match &mut self.target_state {
+            TargetVoteState::V3(v3) => {
+                v3.process_next_vote_slot(next_vote_slot, epoch, current_slot)
+            }
+        }
+    }
+
+    pub fn deinitialize(&mut self) {
+        match &mut self.target_state {
+            TargetVoteState::V3(v3) => *v3 = VoteStateV3::default(),
+        }
+    }
```

Comment:
nit: could also approach this like `init_vote_account_state` and avoid the mutation